api/v1alpha4/virtualmachine_types.go

```diff
@@ -457,6 +457,88 @@ type VirtualMachineCryptoSpec struct {
 	UseDefaultKeyProvider *bool `json:"useDefaultKeyProvider,omitempty"`
 }
 
+// +kubebuilder:validation:Enum=disk;network;cdrom
+
+// VirtualMachineBootOptionsBootableDevice represents the type of bootable device
+// that a VM may be booted from.
+type VirtualMachineBootOptionsBootableDevice string
+
+const (
+	VirtualMachineBootOptionsBootableDiskDevice    VirtualMachineBootOptionsBootableDevice = "disk"
```

Comment:
Kube API conventions wants enums to be CamelCased, ex. `Disk`, `Network`, `CDRom`.